Tests/NIOExtrasTests/LineBasedFramceDecoderTest.swift

```diff
@@ -0,0 +1,155 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import XCTest
+import NIO
+import NIOExtras
+
+class LineBasedFrameDecoderTest: XCTestCase {
+    
+    func testDecodeOneCharacterAtATime() throws {
+        let channel = EmbeddedChannel()
```

Comment:
the setup (create channel & handler, add to pipeline) and tear down (finish) looks always the same, should we move that to a `func setup()`/`func tearDown`?